src/measure_theory/covering/differentiation.lean

```diff
@@ -0,0 +1,723 @@
+    ... = 0 : by rw [ρo, mul_zero] },
+  obtain ⟨u, u_anti, u_pos, u_lim⟩ :
+    ∃ (u : ℕ → ℝ≥0), strict_anti u ∧ (∀ (n : ℕ), 0 < u n) ∧ tendsto u at_top (𝓝 0) :=
+      exists_seq_strict_anti_tendsto (0 : ℝ≥0),
+  have B : ∀ᵐ x ∂μ, ∀ n, ∀ᶠ a in v.filter_at x, ρ a < u n * μ a :=
+    ae_all_iff.2 (λ n, A (u n) (u_pos n)),
+  filter_upwards [B, v.ae_eventually_measure_pos],
+  assume x hx h'x,
+  refine tendsto_order.2 ⟨λ z hz, (ennreal.not_lt_zero hz).elim, λ z hz, _⟩,
+  obtain ⟨w, w_pos, w_lt⟩ : ∃ (w : ℝ≥0), (0 : ℝ≥0∞) < w ∧ (w : ℝ≥0∞) < z :=
+    ennreal.lt_iff_exists_nnreal_btwn.1 hz,
+  obtain ⟨n, hn⟩ : ∃ n, u n < w :=
+    ((tendsto_order.1 u_lim).2 w (ennreal.coe_pos.1 w_pos)).exists,
+  filter_upwards [hx n, h'x, v.eventually_measure_lt_top x],
+  assume a ha μa_pos μa_lt_top,
+  rw ennreal.div_lt_iff (or.inl μa_pos.ne') (or.inl μa_lt_top.ne),
+  exact ha.trans_le (ennreal.mul_le_mul ((ennreal.coe_le_coe.2 hn.le).trans w_lt.le) le_rfl)
+end
+
+section absolutely_continuous
+
+variable (hρ : ρ ≪ μ)
+include hρ
+
+/-- A set of points `s` satisfying both `ρ a ≤ c * μ a` and `ρ a ≥ d * μ a` at arbitrarily small
+sets in a Vitali family has measure `0` if `c < d`. Indeed, the first inequality should imply
+that `ρ s ≤ c * μ s`, and the second one that `ρ s ≥ d * μ s`, a contradiction if `0 < μ s`. -/
+theorem null_of_frequently_le_of_frequently_ge {c d : ℝ≥0} (hcd : c < d) (s : set α)
+  (hc : ∀ x ∈ s, ∃ᶠ a in v.filter_at x, ρ a ≤ c * μ a)
+  (hd : ∀ x ∈ s, ∃ᶠ a in v.filter_at x, (d : ℝ≥0∞) * μ a ≤ ρ a) :
+  μ s = 0 :=
+begin
+  apply null_of_locally_null s (λ x hx, _),
+  obtain ⟨o, xo, o_open, μo⟩ : ∃ o : set α, x ∈ o ∧ is_open o ∧ μ o < ∞ :=
+    measure.exists_is_open_measure_lt_top μ x,
+  refine ⟨o, mem_nhds_within_of_mem_nhds (o_open.mem_nhds xo), _⟩,
+  let s' := s ∩ o,
+  by_contra,
+  apply lt_irrefl (ρ s'),
+  calc ρ s' ≤ c * μ s' : v.measure_le_of_frequently_le (c • μ) hρ s' (λ x hx, hc x hx.1)
+  ... < d * μ s' : begin
+    apply (ennreal.mul_lt_mul_right h _).2 (ennreal.coe_lt_coe.2 hcd),
+    exact (lt_of_le_of_lt (measure_mono (inter_subset_right _ _)) μo).ne,
+  end
+  ... ≤ ρ s' : v.measure_le_of_frequently_le ρ
+    ((measure.absolutely_continuous.refl μ).smul d) s' (λ x hx, hd x hx.1)
+end
+
+/-- If `ρ` is absolutely continuous with respect to `μ`, then for almost every `x`,
+the ratio `ρ a / μ a` converges as `a` shrinks to `x` along a Vitali family for `μ`. -/
+theorem ae_tendsto_div :
+  ∀ᵐ x ∂μ, ∃ c, tendsto (λ a, ρ a / μ a) (v.filter_at x) (𝓝 c) :=
+begin
+  let w : set ℝ≥0∞ := {x | ∃ a : ℚ, x = ennreal.of_real a},
+  have w_count : countable w,
+  { have : w = range (λ (a : ℚ), ennreal.of_real a),
+      by { ext x, simp only [eq_comm, mem_range, mem_set_of_eq] },
+    rw this,
+    exact countable_range _ },
+  have w_dense : dense w,
+  { refine dense_iff_forall_lt_exists_mem.2 (λ c d hcd, _),
+    rcases ennreal.lt_iff_exists_rat_btwn.1 hcd with ⟨q, hq⟩,
+    exact ⟨ennreal.of_real q, ⟨q, rfl⟩, hq.2⟩ },
```

Comment:
Do we have a way to get a countable dense subset, without building it explicitly using ℚ ?

Maybe you use more specific properties of `w` than those two? In that case, could the set with those additional properties be given by a generally useful lemma?

Reply:
I need a countable dense set made of finite numbers. The rationals are just the simplest set satisfying this property. I have clarified this in a comment.

Reply:
We have `exists_countable_dense_bot_top` in `topology.bases` to obtain a countable dense subset which contains bot and top (when they exist). You could add one similar lemma to obtain a set that does not contain them.

Reply:
Done (in the new PR 10405, on which this one depends now).